backend/src/core/services/auth.service.ts

```diff
@@ -11,7 +11,7 @@ import { Transaction } from 'sequelize/types'
 
 export interface AuthService {
   canSendOtp(email: string): Promise<void>
-  sendOtp(email: string, ipAddress: string): Promise<boolean>
+  sendOtp(email: string, ipAddress: string): Promise<string | void>
```

Comment:
Hmm, I was relying on the type system to flag inconsistencies. Wonder why this wasn't flagged?

Reply:
Actually in TypeScript, for application code, it's generally not a great idea to explicitly annotate the return type and better to rely on the type inference, with the exception being if your type is very complex. 

See [this](https://www.youtube.com/watch?v=I6V2FkW1ozQ) for a great explainer.

Reply:
haha this is what I meant by the legacy code in Postman being quite poorly written TypeScript, esp since it was converted from a JavaScript codebase (not helped by the fact that the ppl who did the conversion weren't super familiar with TS imo)

Reply:
the TLDR here is annotating return type allows you to "lie", i.e. it won't throw an error as long as your return type doesn't **violate** the constraint (even if it might be looser than the constraint). See [this excerpt](https://www.youtube.com/watch?v=I6V2FkW1ozQ&t=200s).

This is also why `void` is kinda weird. It doesn't correspond to a type in JS and is closer to telling the callsite "this function might return something, but you're supposed to just ignore it".